src/Domain/CoordinateMaps/Frustum.cpp

```diff
@@ -0,0 +1,339 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/Frustum.hpp"
+
+#include "DataStructures/MakeWithValue.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+
+namespace CoordinateMaps {
+Frustum::Frustum(const std::array<std::array<double, 2>, 4> face_vertices,
```

Comment:
face_vertices to two arrays of arrays, one for upper and one for lower faces orthogonal to the direction of the frustum